src/server/scripts/Spells/spell_warrior.cpp

```diff
@@ -91,6 +97,112 @@ enum WarriorMisc
     SPELL_VISUAL_BLAZING_CHARGE = 26423
 };
 
+// 152278 - Anger Management
+class spell_warr_anger_management_proc : public AuraScript
+{
+    static bool ValidateSpec(ProcEventInfo const& eventInfo, ChrSpecialization spec)
+    {
+        if (Player const* player = eventInfo.GetActor()->ToPlayer())
+            return player->GetPrimarySpecialization() == spec;
+
+        return false;
+    }
+
+    static int32 GetRageSpellCost(SpellInfo const* spellInfo)
+    {
+        for (SpellPowerEntry const* powerCost : spellInfo->PowerCosts)
+            if (powerCost && powerCost->PowerType == POWER_RAGE)
+                return powerCost->ManaCost / 10;
+
+        return 0;
+    }
+
+    static bool CheckArmsProc(AuraEffect const* /*aurEff*/, ProcEventInfo const& eventInfo)
+    {
+        return ValidateSpec(eventInfo, ChrSpecialization::WarriorArms);
+    }
+
+    static bool CheckFuryProc(AuraEffect const* /*aurEff*/, ProcEventInfo const& eventInfo)
+    {
+        return ValidateSpec(eventInfo, ChrSpecialization::WarriorFury);
+    }
+
+    static bool CheckProtectionProc(AuraEffect const* /*aurEff*/, ProcEventInfo const& eventInfo)
+    {
+        return ValidateSpec(eventInfo, ChrSpecialization::WarriorProtection);
+    }
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo(
+        {
+            SPELL_WARRIOR_COLOSSUS_SMASH,
+            SPELL_WARRIOR_BLADESTORM,
+            SPELL_WARRIOR_RAVAGER,
+            SPELL_WARRIOR_WARBREAKER,
+            SPELL_WARRIOR_RECKLESSNESS,
+            SPELL_WARRIOR_AVATAR,
+            SPELL_WARRIOR_SHIELD_WALL
+        });
+    }
+
+    int32 HandleRageSpend(int32 rageAmount, int32 effectAmount)
+    {
+        _rageSpent += rageAmount;
+
+        int32 const effectMultiplier = _rageSpent / effectAmount;
+        _rageSpent = _rageSpent % effectAmount;
+        return effectMultiplier;
+    }
+
+    void HandleProc(AuraEffect const* aurEff, ProcEventInfo const& eventInfo, std::array<int32, 3> const& spellIds)
+    {
+        int32 const rageCost = GetRageSpellCost(eventInfo.GetSpellInfo());
+        int32 const multiplier = HandleRageSpend(rageCost, aurEff->GetAmount());
+
+        for (int32 const spellId : spellIds)
+            if (spellId)
```

Comment:
if we're down to comparing unsigned integers, it's best to compare it as such, not as bool for clarity (if spellId > 0, != 0, etc).